### PR TITLE
Added resetStore Option

### PR DESCRIPTION
**Pull request by Benny739**
Added an option, that the store gets resettet before the sync starts. In some cases the store can show deleted docs (#64).

### CODE REVIEW

File: projects/akita-ng-fire/src/lib/utils/types.ts [hunk 1]
```diff
@@ -26,6 +26,7 @@ export interface WriteOptions extends PathParams {
 export interface SyncOptions extends PathParams {
   storeName: string;
   loading: boolean;
+  reset: boolean;
```
Comment (GrandSchtroumpf) [inline projects/akita-ng-fire/src/lib/utils/types.ts]: Could you add a comment above this params to explain its purpose to developers ? (with JSDoc synthax `/** Will reset the store on .... */`)

File: projects/akita-ng-fire/src/lib/utils/sync-from-action.ts [hunk 2]
```diff
@@ -10,6 +10,15 @@ export function setLoading(storeName: string, loading: boolean) {
   });
 }
 
+/** Set the loading parameter of a specific store */
```
Comment (GrandSchtroumpf) [inline projects/akita-ng-fire/src/lib/utils/sync-from-action.ts]: I think the comment is not related to the function

File: projects/akita-ng-fire/src/lib/collection/collection.service.ts [hunk 3]
```diff
@@ -194,6 +195,10 @@ export class CollectionService<S extends EntityState<EntityType, string>, Entity
 
     const storeName = getStoreName(this.store, syncOptions);
 
+    if (syncOptions.reset) {
```
Comment (GrandSchtroumpf) [inline projects/akita-ng-fire/src/lib/collection/collection.service.ts]: Could you add it to syncCollectionGroup also ?
Does it make sense to add it to `syncDoc` ?

**Comment by GrandSchtroumpf** [comment]
Thanks for the PR. Could you update the documentation to explain developers how to use this option ?

File: projects/akita-ng-fire/src/lib/utils/types.ts (hunk 1 above)
Comment (Benny739) [reply projects/akita-ng-fire/src/lib/utils/types.ts]: Done

File: projects/akita-ng-fire/src/lib/utils/sync-from-action.ts (hunk 2 above)
Comment (Benny739) [reply projects/akita-ng-fire/src/lib/utils/sync-from-action.ts]: you're right, I fixed this mistake

File: projects/akita-ng-fire/src/lib/collection/collection.service.ts (hunk 3 above)
Comment (Benny739) [reply projects/akita-ng-fire/src/lib/collection/collection.service.ts]: You're right, I added the reset Option to syncCollectionGroup, syncManyDocs and to syncDocs.

**Comment by Benny739** [comment]
Hi,

I corrected everything as you requested it.

**Comment by GrandSchtroumpf** [comment]
Ok thx. Keep it mind that it can impact the guard with awaitSync in a case like this : 
```typescript
class Guard extends CollectionGuard {
  // Wait for the data only if there is nothing in the store
  awaitSync() {
    return this.query.getCount() === 0;
  }
}
```
- Go to the movie list page (await before entering as this the the first page).
- Go in another page (store is still filled).
- Go back to the movie list page (not await before entering because store is not reset yet, BUT data won't be available in the store when page is loaded as store has been reset).

**Pull request merged**